consensus/src/round_manager.rs

```diff
@@ -354,14 +355,31 @@ impl RoundManager {
         &mut self,
         new_round_event: NewRoundEvent,
     ) -> anyhow::Result<()> {
+        let is_valid_proposer = self
+            .proposer_election
+            .is_valid_proposer(self.proposal_generator.author(), new_round_event.round);
+
         counters::CURRENT_ROUND.set(new_round_event.round as i64);
         counters::ROUND_TIMEOUT_MS.set(new_round_event.timeout.as_millis() as i64);
         match new_round_event.reason {
             NewRoundReason::QCReady => {
                 counters::QC_ROUNDS_COUNT.inc();
             },
-            NewRoundReason::Timeout(_) => {
+            NewRoundReason::Timeout(ref reason) => {
                 counters::TIMEOUT_ROUNDS_COUNT.inc();
+                counters::ROUND_TIMEOUT_REASON
+                    .with_label_values(&[&reason.to_string(), &is_valid_proposer.to_string()])
```

Comment:
why is this useful? this is previous round timeout?

Reply:
Yes, this is the reason as aggregated by the node. Also, the boolean second label is to see what the next leader thinks the reason is vs others. Ideally, I want everyone to align on the reason, but let's see how varying it is.